.github/workflows/aws-deploy.yml

```diff
@@ -9,6 +9,13 @@ on:
   push:
     branches: ['*']
 
+env:
+  CONTEXT: ${{ github.REF_NAME == 'prod' && 'prod' || 'dev' }}
+  ROLE_TO_ASSUME: >
+    ${{ github.REF_NAME == 'prod' &&
+    'arn:aws:iam::<production-account-number>:role/<production-role-name>' ||
+    'arn:aws:iam::<dev-account-number>:role/<dev-role-name>' }}
```

Comment:
same as previous, not sure if this will work but this is a bit trickier.

How about we use the [branch-names action](https://github.com/marketplace/actions/branch-names) to set roles conditionally depending on branch name?

Reply:
Can you explain more? In the example they use `branch-names` to `echo` one of two messages.  How would you use the `branch-names ``action to set  the`CONTEXT` env
 var' to the right value in both the `tests` and `cdk-deploy` jobs?

Reply:
ok, maybe we don't need the branch-names action but this is what I was thinking which uses the same concept of setting env vars in the steps.

```
name: aws-deploy

on:
  pull_request:
    branches: ['*']
  push:
    branches: ['*']

env:
  ROLE_TO_ASSUME_DEV: 'arn:aws:iam::<production-account-number>:role/<production-role-name>'
  ROLE_TO_ASSUME_PROD: 'arn:aws:iam::<dev-account-number>:role/<dev-role-name>'

concurrency: ci-${{ github.ref }}

jobs:
  tests:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout repository
        uses: actions/checkout@v3

      - name: Static Analysis
        uses: pre-commit/action@v3.0.0

      - name: cdk synth
        uses: youyo/aws-cdk-github-actions@v2
        with:
          cdk_subcommand: 'synth'
          actions_comment: false
          cdk_args: '--context env=${{ env.CONTEXT }} --debug'

      - name: Install dependencies
        run: pip install -r requirements.txt -r requirements-dev.txt

      - name: Run unit tests
        run: python -m pytest tests/ -s -v

  cdk-deploy:
    runs-on: ubuntu-latest
    if: ${{ github.ref == 'refs/heads/main' || github.ref == 'refs/heads/prod'}}
    needs: tests
    permissions:
      id-token: write
      contents: read
      packages: write

    steps:
      
     - name: Configuration for deb branch
       if: ${{ github.ref == 'refs/heads/dev' }}
       run: |
         echo "CONTEXT=dev" >> $GITHUB_ENV
         echo "ROLE_TO_ASSUME=${{ env.ROLE_TO_ASSUME_DEV }}" >> $GITHUB_ENV    

     - name: Configuration for prod branch
       if: ${{ github.ref == 'refs/heads/prod' }}
       run: |
         echo "CONTEXT=prod" >> $GITHUB_ENV
         echo "ROLE_TO_ASSUME=${{ env.ROLE_TO_ASSUME_PROD }}" >> $GITHUB_ENV    
    
     - name: Checkout repository
        uses: actions/checkout@v3

     - name: Assume AWS Role
       uses: aws-actions/configure-aws-credentials@v1
       with:
         aws-region: us-east-1
         role-to-assume: ${{env.ROLE_TO_ASSUME }}
         role-session-name: GitHubActions-${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
         role-duration-seconds: 1200

     - name: cdk deploy
       uses: youyo/aws-cdk-github-actions@v2
       with:
         cdk_subcommand: 'deploy'
         cdk_args: '--require-approval never --context env=${{ env.CONTEXT }} --progress events --debug'
         actions_comment: false
```